internal/gatewayapi/route.go

```diff
@@ -233,6 +233,7 @@ func (t *Translator) processHTTPRouteRules(httpRoute *HTTPRouteContext, parentRe
 				BackendRef: &rule.BackendRefs[i].BackendRef,
 				Filters:    rule.BackendRefs[i].Filters,
 			}
+			// ds will never be nil here because processDestination returns an empty DestinationSetting for invalid backendRefs.
```

Comment:
isnt `ds` nil when weight is 0 ?

Reply:
I changed the logic to also return an emptyDS with weight as 0 to align the error logic. 

It's quite confusing when the returned DS is nil or not.